vsutil.py

```diff
@@ -87,11 +87,17 @@ def get_plane_size(frame: Union[vs.VideoFrame, vs.VideoNode], /, planeno: int) -
     return width, height
 
 
-def iterate(base: T, function: Callable[[T], T], count: int) -> T:
+def iterate(base: T, function: Callable[[Union[T, R]], R], count: int) -> Union[T, R]:
     """
     Utility function that executes a given function a given number of times.
     """
-    return reduce(lambda v, _: function(v), range(count), base)
+    if count < 0:
+        raise ValueError('Count cannot be negative.')
+
+    v: Union[T, R] = base
```

Comment:
Did the `reduce` implementation break the typechecker or what’s the reason for switching to an imperative approach?

Reply:
It requires the `v: Union[T, R] = base` line and I personally considered it to be less readable (and so did @OrangeChannel when we talked about it).

Reply:
Understandable.